pkg/kubelet/server/server.go

```diff
@@ -417,6 +419,20 @@ func (s *Server) InstallDebuggingHandlers(criHandler http.Handler) {
 	}
 }
 
+// InstallDebuggingDisabledHandlers registers the HTTP request patterns that provide better error message
+func (s *Server) InstallDebuggingDisabledHandlers() {
+	glog.Infof("Adding debug disabled handlers to kubelet server.")
+
+	h := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		http.Error(w, "Server endpoints for log collection and local running of containers and commands are disabled.", http.StatusMethodNotAllowed)
+	})
+
+	paths := []string{"/run/", "/exec/", "/attach/", "/portForward/", "/containerLogs/", "/runningpods/"}
+	for _, p := range paths {
+		s.restfulCont.Handle(p, h)
```

Comment:
Does this also handle sub paths? E.g. `/run/foo`?

Reply:
Yes, we add path by a string like "/run/" with a "/" at the end. It should be able to take care of sub paths.

And, I tested with "kubectl  exec" commands, it actually doing post to path "/exec/${namespace}/${pod}/${container}?command=${cmd}...", it works as expected.